pydantic/funcs.py

```diff
@@ -0,0 +1,226 @@
+    merged_config: CoreConfig = {
+        **(config or {}),  # type: ignore[misc]
+        **getattr(__type, 'model_config', {}),
+    }
+    arbitrary_types = bool((config or {}).get('arbitrary_types_allowed', False))
+    local_ns = _typing_extra.parent_frame_namespace(parent_depth=parent_depth)
+    # BaseModel uses it's own __module__ to find out where it was defined
+    # and then look for symbols to resolve forward references in those globals
+    # On the other hand this function can be called with arbitrary objects,
+    # including type aliases where __module__ (always `typing.py`) is not useful
+    # So instead we look at the globals in our parent stack frame
+    # This works for the case where this function is called in a module that
+    # has the target of forward references in its scope but
+    # does not work for more complex cases
+    # for example, take the following:
+    #
+    # a.py
+    # ```python
+    # from typing import List, Dict
+    # IntList = List[int]
+    # OuterDict = Dict[str, 'IntList']
+    # ```
+    #
+    # b.py
+    # ```python
+    # from pydantic import funcs
+    # from a import OuterDict
+    # IntList = int  # replaces the symbol the forward reference is looking for
+    # v = funcs.analyze_type(OuterDict)
+    # v({"x": 1})  # should fail but doesn't
+    # ```
+    #
+    # If OuterDict were a BaseModel this would work because it would resolve
+    # the forward reference within the `a.py` namespace.
+    # But `pydantic.funcs.analyze_type(OuterDict)`
+    # can't know what module OuterDict came from.
+    # In other words, the assumption that _all_ forward references exist in the
+    # module we are being called from is not technically always true
+    # Although most of the time it is and it works fine for recursive models and such/
+    # BaseModel's behavior isn't perfect either and _can_ break in similar ways,
+    # so there is no right or wrong between the two.
+    # But at the very least this behavior is _subtly_ different from BaseModel's.
+    global_ns = sys._getframe(max(parent_depth - 1, 1)).f_globals.copy()
+    global_ns.update(local_ns or {})
+    gen = _generate_schema.GenerateSchema(arbitrary_types=arbitrary_types, types_namespace=global_ns, typevars_map={})
+    return AnalyzedType[Any](gen.generate_schema(__type), merged_config)
+
+
+def json_schema(
+    __types_or_schema: AnalyzedType[Any] | type[Any],
+    *,
+    by_alias: bool = True,
+    ref_template: str = DEFAULT_REF_TEMPLATE,
+    schema_generator: type[GenerateJsonSchema] = GenerateJsonSchema,
+) -> dict[str, Any]:
+    schema_generator_instance = schema_generator(by_alias=by_alias, ref_template=ref_template)
+
+    core_schema = (
+        __types_or_schema.core_schema
+        if isinstance(__types_or_schema, AnalyzedType)
+        else analyze_type(__types_or_schema).core_schema
+    )
+
+    return schema_generator_instance.generate(core_schema)
+
+
+def json_schemas(
```

Comment:
this is a bit awkward now if everything else is living on AnalyzedType. Don't have a great suggestion

Reply:
I think I can move `json_schema` to be an instance method. Maybe `json_schemas` can be a static method?

Reply:
sounds good to me